### PR TITLE
Add direct include in CommandPathParams.h

### PR DESCRIPTION
**Pull request by wendythewan**
#### Problem
CommandPathParams relies on transitive includes (things defined in a header that is included by another header that is included by it). This is not ideal for "include what you use" rationale (see https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md).

Having "include what you use" makes some integrations easier if their automated tools validate this.

#### Change overview
Add include in CommandPathParams.h

#### Testing
* Compile

### CODE REVIEW

File: src/app/CommandPathParams.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #pragma once
 
 #include <app/util/basic-types.h>
+#include "lib/core/GroupId.h"
```
Comment (mrjerryjohns) [inline src/app/CommandPathParams.h]: `basic-types.h` includes `lib/core/DataModelTypes.h`, which includes `GroupId.h` already. So this isn't necessary.

**Review by mrjerryjohns — DISMISSED** [review]
Change not needed.

File: src/app/CommandPathParams.h (hunk 1 above)
Comment (mspang) [reply src/app/CommandPathParams.h]: It isn't necessary to compile, but it's necessary to satisfy IWYU checkers (which includes our internal tools). There is no harm, we should land changes like this if they help SDK consumers satisfy their own tooling needs.

**Review by mspang — APPROVED** [review]

File: src/app/CommandPathParams.h (hunk 1 above)
Comment (andy31415) [reply src/app/CommandPathParams.h]: Include what you use seems a reasonable requirement. We should not rely on transitive includes.

File: src/app/CommandPathParams.h [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #pragma once
 
 #include <app/util/basic-types.h>
+#include <lib/core/GroupId.h>
```
Comment (bzbarsky-apple) [inline src/app/CommandPathParams.h:22]: Why?  basic-types.h includes `lib/core/DataModelTypes.h` which includes `lib/core/GroupId.h`.

Including it directly here does not hurt, but why was it needed?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/CommandPathParams.h (hunk 2 above)
Comment (wendythewan) [reply src/app/CommandPathParams.h:22]: It is not needed for compilation so "missing" in the header isn't quite right really.

**Comment by mrjerryjohns** [comment]
The updated rationale provided by Michael makes sense - please include that in the original description next time to provide context to reviewers.

**Review by mrjerryjohns — APPROVED** [review]

File: src/app/CommandPathParams.h (hunk 2 above)
Comment (andy31415) [reply src/app/CommandPathParams.h:22]: @bzbarsky-apple - we had the same comment from @mrjerryjohns . The reason is "include what you use" checkers that force not to rely on inter-include guarantees that may change in time.

So agree not needed for compiles, however it makes some cleanliness sense (this is why include what you use checkers exist) and it makes some users of the SDK have an easier life to integrate.

Comment (andy31415) [reply src/app/CommandPathParams.h:22]: https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md is the rationale I believe.

Comment (andy31415) [reply src/app/CommandPathParams.h:22]: I updated summary based on my understanding of the patch: it is not that the include is missing (it is not strictly needed) but that we want to IWYU.

Comment (bzbarsky-apple) [reply src/app/CommandPathParams.h:22]: Yes, from an IWYU pov this makes total sense.  It does mean headers like DataModelTypes.h are not that useful... ;)

**Pull request merged**